src/Grpc.AspNetCore.Server/GrpcEndpointRouteBuilderExtensions.cs

```diff
@@ -43,11 +45,44 @@ public static class GrpcEndpointRouteBuilderExtensions
         ValidateServicesRegistered(builder.ServiceProvider);
 
         var serviceRouteBuilder = builder.ServiceProvider.GetRequiredService<ServiceRouteBuilder<TService>>();
-        var endpointConventionBuilders = serviceRouteBuilder.Build(builder);
+        var endpointConventionBuilders = serviceRouteBuilder.Build(builder, argument: null);
 
         return new GrpcServiceEndpointConventionBuilder(endpointConventionBuilders);
     }
 
+    /// <summary>
+    /// Maps incoming requests to the specified <see cref="ServerServiceDefinition"/> instance.
+    /// </summary>
+    /// <param name="builder">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+    /// <param name="serviceDefinition">The instance of <see cref="ServerServiceDefinition"/>.</param>
+    /// <returns>A <see cref="GrpcServiceEndpointConventionBuilder"/> for endpoints associated with the service.</returns>
+    public static GrpcServiceEndpointConventionBuilder MapGrpcService(this IEndpointRouteBuilder builder, ServerServiceDefinition serviceDefinition)
+    {
+        ArgumentNullException.ThrowIfNull(builder, nameof(builder));
+        ArgumentNullException.ThrowIfNull(serviceDefinition, nameof(serviceDefinition));
+
+        var serviceRouteBuilder = builder.ServiceProvider.GetRequiredService<ServiceRouteBuilder<ServerServiceDefinitionMarker>>();
```

Comment:
Any way to share method internals with the already existing one?

If not, should the new method also do `ValidateServicesRegistered(builder.ServiceProvider);`